### PR TITLE
feat(expressions): add page base routing

### PR DESCRIPTION
**Pull request by Shurtu-gal**
### Your checklist for this pull request

🚨Please review the [guidelines for contributing](https://github.com/Monday-Morning/project-tahiti/blob/main/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch** (left side). Also you should start _your branch_ off _default branch_.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [ ] I have added necessary documentation (if appropriate)

### Linked Issues
(feat: add page base routing of subcategories #289)

Fixes

### Description

Please describe your pull request.

❤️Thank you!

### Post merge checklist

- [x] Follow steps from the [guidelines for contributing](https://github.com/Monday-Morning/project-tahiti/blob/main/CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

File: client/src/pages/expressions/[...subCategory].jsx [hunk 1]
```diff
@@ -21,33 +21,29 @@ const EditorialPage = ({
...
   articleList,
   countOfArticles,
+  pageNumber,
 }) => {
-  const { isFallback } = useRouter();
+  const { isFallback, push } = useRouter();
 
-  const [pageNo, setPageNo] = useState(1);
+  const [pageNo, setPageNo] = useState(pageNumber);
   const [articleLists, setArticleLists] = useState(articleList);
```
Comment (120EE0692) [inline client/src/pages/expressions/[...subCategory].jsx:29]: @Shurtu-gal is this anymore required and setArticleLists is no more used so consider removing this.

**Review by 120EE0692 — CHANGES_REQUESTED** [review]

File: client/src/pages/expressions/[...subCategory].jsx (hunk 1 above)
Comment (Shurtu-gal) [reply client/src/pages/expressions/[...subCategory].jsx:29]: Sure

File: client/src/pages/expressions/[...subCategory].jsx [hunk 2]
```diff
@@ -21,33 +21,29 @@ const EditorialPage = ({
...
   countOfArticles,
+  pageNumber,
 }) => {
-  const { isFallback } = useRouter();
+  const { isFallback, push } = useRouter();
 
-  const [pageNo, setPageNo] = useState(1);
-  const [articleLists, setArticleLists] = useState(articleList);
+  const [pageNo, setPageNo] = useState(pageNumber);
+  const [articleLists] = useState(articleList);
```
Comment (120EE0692) [inline client/src/pages/expressions/[...subCategory].jsx:29]: @Shurtu-gal remove useState as well.

**Review by 120EE0692 — CHANGES_REQUESTED** [review]
on clicking seeAll button it should go to expression/[subcategory]/1

**Review by 120EE0692 — APPROVED** [review]

File: client/src/pages/expressions/[...subCategory].jsx [hunk 3]
```diff
@@ -121,13 +112,13 @@ const EditorialPage = ({
           content='Monday Morning is the Media Body of National Institute Of Technology Rourkela. Monday Morning covers all the events, issues and activities going on inside the campus. Monday morning also serves as a link between the administration and the students.'
         />
       </Head>
-      {!isLoading && !isFallback && articleLists ? (
+      {!isLoading && !isFallback ? (
```
Comment (rutajdash) [inline client/src/pages/expressions/[...subCategory].jsx:115]: @120EE0692 do we not have a loading page added here for the fallback?

**Comment by rutajdash** [comment]
@Shurtu-gal @120EE0692 please confirm this code does pass eslint and prettier checks.

**Pull request merged**